csi/node_server.go

```diff
@@ -899,6 +860,86 @@ func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 	return &csi.NodeExpandVolumeResponse{CapacityBytes: requestedSize}, nil
 }
 
+// prepareAndExpandEncryptedVolume handles the expansion of encrypted volumes and
+// returns volume is a block volume, updated device path, and error if any.
+func (ns *NodeServer) prepareAndExpandEncryptedVolume(req *csi.NodeExpandVolumeRequest, volume *longhornclient.Volume, volumeID, devicePath string) (bool, string, error) {
+	isBlockVolume := req.VolumeCapability.GetBlock() != nil
+	if isBlockVolume {
+		log.Debugf("Volume %v on node %v does not require filesystem resize since it is access mode Block", volumeID, ns.nodeID)
+	}
+
+	if !volume.Encrypted {
+		return isBlockVolume, devicePath, nil
+	}
+
+	diskFormat, err := ns.getDiskFormat(devicePath)
+	if err != nil {
+		return isBlockVolume, devicePath, err
+	}
```

Comment:
Move to outside of the function.
The function is to expand a encrypted volume. It doesn't need to check if the volume is encrypted or not.